src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -925,6 +926,71 @@ void LoadHandler::parserEndElement(GMarkupParseContext* context, const gchar* el
     }
 }
 
+void LoadHandler::fixNullPressureValues() {
+    /*
+     * Due to various bugs (see e.g. https://github.com/xournalpp/xournalpp/issues/3643), old files may contain strokes
+     * with non-positive pressure values.
+     *
+     * Those strokes thus fails the somewhat reasonable assumption that pressure values should be positive.
+     * The portions of stroke with non-positive pressure values are in any case invisible.
+     *
+     * This function fixes corrupted strokes by
+     *  1- removing every point with a non-positive pressure value.
+     *  2- if required, splitting the affected stroke into several strokes.
+     *  3- entirely deleting strokes without any valid pressure value
+     */
+    auto& pts = stroke->getPointVector();
+    if (pressureBuffer.size() >= pts.size()) {
+        // Too many pressure values. Drop the last ones
+        assert(pts.size() >= 2);  // An error has already been returned otherwise
+        pressureBuffer.resize(pts.size() - 1);
+    }
+
+    auto nextPositive = std::find_if(pressureBuffer.begin(), pressureBuffer.end(), [](double v) { return v > 0; });
+
+    std::vector<std::vector<Point>> strokePortions;
+    while (nextPositive != pressureBuffer.end()) {
```

Comment:
nit: could probably be simplified into a `do { auto nextPositive = ...; } while (nextPositive != pressureBuffer.end()) { ... }` loop.